src/dist/replication/common/replication_common.h

```diff
@@ -51,7 +51,6 @@ class replication_options
     bool verbose_commit_log_on_start;
     bool delay_for_fd_timeout_on_start;
     bool empty_write_disabled;
-    bool allow_non_idempotent_write;
```

Comment:
We used `allow_non_idempotent_write` to config whether a cluster support non-idempotent operation. Removing this option means all clusters will not restrict non-idempotent operation. Is it safe to do this? Is there other condition that should not coexistent with non-idempotent except duplication?

Reply:
> Is there other condition that should not coexistent with non-idempotent except duplication?

No. If there is, we should forbid them all.